libcaf_net/caf/net/transport_worker_dispatcher.hpp

```diff
@@ -65,48 +68,48 @@ class transport_worker_dispatcher {
 
   template <class Parent>
   error handle_data(Parent& parent, span<byte> data, id_type id) {
-    auto it = workers_by_id_.find(id);
-    if (it == workers_by_id_.end()) {
-      // TODO: where to get node_id from here?
+    auto worker = find_by_id(id);
+    if (!worker) {
+      // TODO: where to get id_type from here?
       add_new_worker(parent, node_id{}, id);
-      it = workers_by_id_.find(id);
+      worker = find_by_id(id);
     }
-    auto worker = it->second;
     return worker->handle_data(parent, data);
   }
 
   template <class Parent>
   void write_message(Parent& parent,
                      std::unique_ptr<endpoint_manager_queue::message> msg) {
-    auto sender = msg->msg->sender;
-    if (!sender)
+    auto receiver = msg->receiver;
+    if (!receiver)
       return;
-    auto nid = sender->node();
-    auto it = workers_by_node_.find(nid);
-    if (it == workers_by_node_.end()) {
+    auto nid = receiver->node();
+    auto worker = find_by_node(nid);
+    if (!worker) {
       // TODO: where to get id_type from here?
       add_new_worker(parent, nid, id_type{});
-      it = workers_by_node_.find(nid);
+      worker = find_by_node(nid);
     }
-    auto worker = it->second;
     worker->write_message(parent, std::move(msg));
   }
 
   template <class Parent>
-  void write_packet(Parent& parent, span<const byte> header,
-                    span<const byte> payload, id_type id) {
-    parent.write_packet(header, payload, id);
+  void resolve(Parent& parent, const uri& locator, const actor& listener) {
+    if (auto worker = find_by_node(make_node_id(locator)))
+      worker->resolve(parent, locator.path(), listener);
```

Comment:
This lacks an else tree. `listener` will wait forever on a message on error.

Reply:
There is no way to return an error from `resolve` - But I can add that.

Reply:
I don't mean *returning* an error. I mean sending an error to the listener. :)

Reply:
How exactly though? I would need to add return type `error` to the whole chain. But what could I do about the `endpoint_manager_impl`?

Reply:
You answered faster than I was typing.. ignore that last comment!